spring-boot-project/spring-boot-autoconfigure/src/test/java/org/springframework/boot/autoconfigure/data/neo4j/Neo4jDataAutoConfigurationTests.java

```diff
@@ -16,257 +16,402 @@
-import org.springframework.data.neo4j.bookmark.BookmarkManager;
-import org.springframework.data.neo4j.mapping.Neo4jMappingContext;
-import org.springframework.data.neo4j.transaction.Neo4jTransactionManager;
-import org.springframework.data.neo4j.web.support.OpenSessionInViewInterceptor;
-import org.springframework.web.context.WebApplicationContext;
+import org.springframework.data.neo4j.core.DatabaseSelection;
+import org.springframework.data.neo4j.core.DatabaseSelectionProvider;
+import org.springframework.data.neo4j.core.Neo4jClient;
+import org.springframework.data.neo4j.core.Neo4jOperations;
+import org.springframework.data.neo4j.core.Neo4jTemplate;
+import org.springframework.data.neo4j.core.ReactiveDatabaseSelectionProvider;
+import org.springframework.data.neo4j.core.ReactiveNeo4jClient;
+import org.springframework.data.neo4j.core.ReactiveNeo4jOperations;
+import org.springframework.data.neo4j.core.ReactiveNeo4jTemplate;
+import org.springframework.data.neo4j.core.convert.Neo4jConversions;
+import org.springframework.data.neo4j.core.transaction.Neo4jTransactionManager;
+import org.springframework.data.neo4j.core.transaction.ReactiveNeo4jTransactionManager;
+import org.springframework.test.util.ReflectionTestUtils;
+import org.springframework.transaction.PlatformTransactionManager;
+import org.springframework.transaction.ReactiveTransactionManager;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+import reactor.test.StepVerifier;
+
+import org.neo4j.driver.Driver;
 
 import static org.assertj.core.api.Assertions.assertThat;
-import static org.mockito.ArgumentMatchers.any;
 import static org.mockito.Mockito.mock;
-import static org.mockito.Mockito.verify;
 
 /**
- * Tests for {@link Neo4jDataAutoConfiguration}. Tests should not use the embedded driver
- * as it requires the complete Neo4j-Kernel and server to function properly.
- *
- * @author Stephane Nicoll
- * @author Michael Hunger
- * @author Vince Bickers
- * @author Andy Wilkinson
- * @author Kazuki Shimizu
- * @author Michael Simons
+ * @author Michael J. Simons
  */
 class Neo4jDataAutoConfigurationTests {
 
-	private WebApplicationContextRunner contextRunner = new WebApplicationContextRunner()
-			.withClassLoader(new FilteredClassLoader(EmbeddedDriver.class))
-			.withUserConfiguration(TestConfiguration.class).withConfiguration(
-					AutoConfigurations.of(Neo4jDataAutoConfiguration.class, TransactionAutoConfiguration.class));
-
-	@Test
-	void defaultConfiguration() {
-		this.contextRunner.withPropertyValues("spring.data.neo4j.uri=http://localhost:8989").run((context) -> {
-			assertThat(context).hasSingleBean(org.neo4j.ogm.config.Configuration.class);
-			assertThat(context).hasSingleBean(SessionFactory.class);
-			assertThat(context).hasSingleBean(Neo4jTransactionManager.class);
-			assertThat(context).doesNotHaveBean(OpenSessionInViewInterceptor.class);
-			assertThat(context).doesNotHaveBean(BookmarkManager.class);
-		});
-	}
+	@Nested
```

Comment:
We don't use `@Nested` anywhere else. I am personally not keen to introduce this style as part of this PR. I also think it makes the diff very hard to read.